index.js

```diff
@@ -143,16 +163,25 @@ class DetailsDialogElement extends HTMLElement {
     if (!details) return
 
     const summary = details.querySelector('summary')
-    if (summary) summary.setAttribute('aria-haspopup', 'dialog')
+    if (summary) {
+      summary.setAttribute('aria-haspopup', 'dialog')
+      summary.addEventListener('click', onSummaryClick, {capture: true})
+    }
 
     details.addEventListener('toggle', toggle)
     state.details = details
   }
 
   disconnectedCallback() {
     const state = initialized.get(this)
-    if (!state || !state.details) return
-    state.details.removeEventListener('toggle', toggle)
+    if (!state) return
+    const {details} = state
+    if (!details) return
+    details.removeEventListener('toggle', toggle)
+    const summary = details.querySelector('summary')
```

Comment:
Wasn't sure if I should store the `summary` in the `state` or it is okay to requery for it here.